kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/util/ExpressionState.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.dmn.client.editors.expressions.util;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.enterprise.event.Event;
+
+import org.kie.workbench.common.dmn.api.definition.HasExpression;
+import org.kie.workbench.common.dmn.api.definition.HasName;
+import org.kie.workbench.common.dmn.api.definition.HasVariable;
+import org.kie.workbench.common.dmn.api.definition.model.Expression;
+import org.kie.workbench.common.dmn.api.definition.model.InformationItemPrimary;
+import org.kie.workbench.common.dmn.api.property.dmn.QName;
+import org.kie.workbench.common.dmn.api.property.dmn.types.BuiltInType;
+import org.kie.workbench.common.dmn.client.editors.expressions.ExpressionEditorView;
+import org.kie.workbench.common.dmn.client.widgets.grid.model.ExpressionEditorChanged;
+
+public class ExpressionState {
+
+    private final HasExpression hasExpression;
+    private final Event<ExpressionEditorChanged> editorSelectedEvent;
+    private final ExpressionEditorView view;
+    private final String nodeUUID;
+    private Optional<Expression> savedExpression;
```

Comment:
@danielzhe I think you don't really need to use `savedExpression` as `Optional` here. This is because, when you use `getSavedExpression`, you set `null` if the optional is empty. Moreover, using an Optional as field and/or method parameter is discouraged.

Reply:
Yes, here I can do that change. Maybe at some point of the coding I was using another logic that required it to be Optional but know you are right, it is not required anymore.

Reply:
Oh, I remember now. I did in this way to be more easy to test it.